Docs/Doxyfile

```diff
@@ -2046,7 +2046,24 @@ INCLUDE_FILE_PATTERNS  =
 # recursively expanded use the := operator instead of the = operator.
 # This tag requires that the tag ENABLE_PREPROCESSING is set to YES.
 
-PREDEFINED             =
+PREDEFINED             = AMREX_Linux=1            \
+                         AMREX_PARTICLES=1        \
+                         AMREX_USE_MPI=1          \
+                         AMREX_USE_OMP=1          \
+                         AMREX_SPACEDIM=3         \
+                         AMREX_TINY_PROFILING=1   \
+                         BL_Linux=1               \
+                         BL_USE_MPI=1             \
+                         BL_USE_OMP=1             \
+                         BL_USE_SENSEI_INSITU=1   \
+                         WARPX=1                  \
+                         WARPX_DIM_RZ=1           \
+                         WARPX_DIM_XZ=1           \
+                         WARPX_DO_ELECTROSTATIC=1 \
+                         WARPX_USE_GPU=1          \
+                         WARPX_USE_OPENPMD=1      \
+                         WARPX_USE_PSATD=1        \
+                         WARPX_USE_PSATD_HYBRID=1
```

Comment:
Some of these options are in principle incompatible (e.g. `WARPX_DIM_RZ=1` and `WARPX_DIM_XZ=1` indicates that the simulation is both cylindrical and Cartesian geometry)
Is that potentially an issue?

Reply:
Yes, I was aware of the potential incompatibility if one would compile that way. The idea behind still defining as much as possible is to expose as many classes as possible to the Doxygen source code parser.
It's not an issue for Doxygen, as long as it does not introduce invalid C++ syntax (runtime logic does not matter here).